app/controllers/users_controller.rb

```diff
@@ -82,8 +81,8 @@ def new_registration
   def create_registration
     @user = current_user
     @user.assign_attributes(user_params)
-    @user.user_attributes[:department] = params[:user][:user_attributes][:department]
-    @user.user_attributes[:school] = params[:user][:user_attributes][:school]
+    @user.department = params[:user][:user_attributes][:department]
+    @user.school = params[:user][:user_attributes][:school]
     @user.user_attributes[:marli_renewal] = params[:user][:user_attributes][:marli_renewal]
     @user.user_attributes[:affiliation_text] = params[:user][:user_attributes][:affiliation_text]
```

Comment:
@hab278 And actually adding this following line to the bottom of `_register_form.html.erb` will probably fix this

```
<%= f.hidden_field :affiliation_text, value: affiliation_text %>
```

And you can remove the lines in that file that look like the following:

```
<%= f.fields_for :user_attributes do |u| %>
...
<% end %>
```

Because all these fields will be top-level.

But then the above lines should obviously look like this:

```ruby
@user.marli_renewal = params[:user][:marli_renewal]
@user.affiliation_text = params[:user][:affiliation_text]
```